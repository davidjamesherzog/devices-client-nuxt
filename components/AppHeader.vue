<template>
  <nav
    class="navbar header has-shadow is-primary"
    role="navigation"
    aria-label="main navigation"
  >
    <div class="navbar-brand">
      <div class="navbar-burger" @click="toggleMenu()">
        <span />
        <span />
        <span />
      </div>
      <a class="navbar-item" href="/">
        <img src="~assets/buefy.png" alt="Buefy" height="28" />
      </a>
    </div>
  </nav>
</template>

<script lang="ts">
import { Component, Vue } from 'nuxt-property-decorator';

@Component({
  name: 'AppHeader'
})
export default class AppHeader extends Vue {
  public toggleMenu(): void {
    this.$nuxt.$emit('toggle-menu');
  }
}
</script>

<style>
.navbar-burger {
  margin-left: 0;
}
</style>
