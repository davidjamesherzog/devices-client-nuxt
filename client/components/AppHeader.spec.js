import { createLocalVue, shallowMount } from '@vue/test-utils';
import Buefy from 'buefy';
import AppHeader from '~/components/AppHeader.vue';

describe('AppHeader', () => {
  describe('toggleMenu', () => {
    test('should emit event', () => {
      const appHeader = new AppHeader();
      appHeader.$nuxt = {
        $emit: jest.fn()
      };
      appHeader.toggleMenu();
      expect(appHeader.$nuxt.$emit).toHaveBeenCalledWith('toggle-menu');
    });
  });

  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('is a Vue instance', () => {
      const wrapper = shallowMount(AppHeader, { localVue });
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
