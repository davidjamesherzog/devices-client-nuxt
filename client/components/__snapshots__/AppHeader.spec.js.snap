// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader template is a Vue instance 1`] = `
<nav role="navigation" aria-label="main navigation" class="app-header navbar header has-shadow is-primary">
  <div class="navbar-brand">
    <div class="navbar-burger"><span></span> <span></span> <span></span></div> <a href="/" class="navbar-item">
      <h4 class="is-size-4">Devices</h4>
    </a>
  </div>
</nav>
`;
